// ng-zorro-antd-mobile/esm2015/segmented-control/segmented-control.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input, Output, HostBinding, EventEmitter } from '@angular/core';
let SegmentedControlComponent = class SegmentedControlComponent {
    constructor() {
        this.prefixCls = 'am-segment';
        this.tintColor = '#2DB7F5';
        this.disabled = false;
        this.selectedIndex = 0;
        this.onChange = new EventEmitter();
        this.role = 'tablist';
        this.amSegment = true;
    }
    get amDisabled() {
        return this.disabled;
    }
    onClick(index, value) {
        if (!this.disabled && index !== this.selectedIndex) {
            this.selectedIndex = index;
            this.onChange.emit({ selectedIndex: index, value: value });
        }
    }
};
__decorate([
    Input(),
    __metadata("design:type", String)
], SegmentedControlComponent.prototype, "tintColor", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], SegmentedControlComponent.prototype, "disabled", void 0);
__decorate([
    Input(),
    __metadata("design:type", Number)
], SegmentedControlComponent.prototype, "selectedIndex", void 0);
__decorate([
    Input(),
    __metadata("design:type", Array)
], SegmentedControlComponent.prototype, "values", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], SegmentedControlComponent.prototype, "onChange", void 0);
__decorate([
    HostBinding('attr.role'),
    __metadata("design:type", Object)
], SegmentedControlComponent.prototype, "role", void 0);
__decorate([
    HostBinding('class.am-segment'),
    __metadata("design:type", Boolean)
], SegmentedControlComponent.prototype, "amSegment", void 0);
__decorate([
    HostBinding('class.am-segment-disabled'),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [])
], SegmentedControlComponent.prototype, "amDisabled", null);
SegmentedControlComponent = __decorate([
    Component({
        selector: 'SegmentedControl, nz-segmented-control',
        template: "<div\n  role=\"tab\"\n  *ngFor=\"let value of values; let i = index\"\n  class=\"{{ prefixCls }}-item\"\n  [ngClass]=\"{ 'am-segment-item-selected': i === selectedIndex }\"\n  [ngStyle]=\"{\n    'border-color': tintColor,\n    color: i === selectedIndex ? '#fff' : tintColor,\n    'background-color': i === selectedIndex ? tintColor : 'transparent'\n  }\"\n  (click)=\"onClick(i, value)\"\n>\n  <div\n    class=\"{{ prefixCls }}-item-inner\"\n    [ngStyle]=\"{ 'background-color': i === selectedIndex ? tintColor : 'transparent' }\"\n  ></div>\n  {{ value }}\n</div>\n"
    }),
    __metadata("design:paramtypes", [])
], SegmentedControlComponent);
export { SegmentedControlComponent };
