// ng-zorro-antd-mobile/esm5/card/card-header.component.js
import { __decorate, __metadata } from "tslib";
import { Component, ViewEncapsulation, Input, HostBinding, TemplateRef } from '@angular/core';
var CardHeaderComponent = /** @class */ (function () {
    function CardHeaderComponent() {
        this.prefixCls = 'am-card-header';
        this.thumb = null;
        this.thumbStyle = null;
        this.title = null;
        this.extra = null;
        this.cardBodyWrapper = true;
    }
    CardHeaderComponent.prototype.isTemplateRef = function (value) {
        return value instanceof TemplateRef;
    };
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], CardHeaderComponent.prototype, "thumb", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], CardHeaderComponent.prototype, "thumbStyle", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], CardHeaderComponent.prototype, "title", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], CardHeaderComponent.prototype, "extra", void 0);
    __decorate([
        HostBinding('class.am-card-header'),
        __metadata("design:type", Boolean)
    ], CardHeaderComponent.prototype, "cardBodyWrapper", void 0);
    CardHeaderComponent = __decorate([
        Component({
            selector: 'CardHeader, nzm-card-header',
            template: "<div class=\"{{ prefixCls }}-content\">\n  <img *ngIf=\"thumb && !isTemplateRef(thumb)\" src=\"{{ thumb }}\" [ngStyle]=\"thumbStyle\" />\n  <ng-container *ngIf=\"thumb && isTemplateRef(thumb)\" [ngTemplateOutlet]=\"thumb\"></ng-container>\n  <ng-container *ngIf=\"!isTemplateRef(title); else titleTemplate\">{{ title }}</ng-container>\n</div>\n<div *ngIf=\"extra\" class=\"{{ prefixCls }}-extra\">\n  <ng-container *ngIf=\"!isTemplateRef(extra); else extraTemplate\">{{ extra }}</ng-container>\n</div>\n<ng-template #titleTemplate>\n  <ng-template [ngTemplateOutlet]=\"title\"></ng-template>\n</ng-template>\n<ng-template #extraTemplate>\n  <ng-template [ngTemplateOutlet]=\"extra\"></ng-template>\n</ng-template>\n",
            encapsulation: ViewEncapsulation.None
        }),
        __metadata("design:paramtypes", [])
    ], CardHeaderComponent);
    return CardHeaderComponent;
}());
export { CardHeaderComponent };
