// ng-zorro-antd-mobile/esm5/card/card.component.js
import { __decorate, __metadata } from "tslib";
import { Component, ViewEncapsulation, Input, HostBinding } from '@angular/core';
var CardComponent = /** @class */ (function () {
    function CardComponent() {
        this.full = false;
        this.cardWrapper = true;
    }
    Object.defineProperty(CardComponent.prototype, "cardFull", {
        get: function () {
            return this.full;
        },
        enumerable: true,
        configurable: true
    });
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], CardComponent.prototype, "full", void 0);
    __decorate([
        HostBinding('class.am-card'),
        __metadata("design:type", Boolean)
    ], CardComponent.prototype, "cardWrapper", void 0);
    __decorate([
        HostBinding('class.am-card-full'),
        __metadata("design:type", Boolean),
        __metadata("design:paramtypes", [])
    ], CardComponent.prototype, "cardFull", null);
    CardComponent = __decorate([
        Component({
            selector: 'Card, nzm-card',
            template: "<ng-content></ng-content>\n",
            encapsulation: ViewEncapsulation.None
        }),
        __metadata("design:paramtypes", [])
    ], CardComponent);
    return CardComponent;
}());
export { CardComponent };
