// ng-zorro-antd-mobile/esm2015/tab-bar/tab-bar-item.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input, ViewChild, TemplateRef } from '@angular/core';
import { TabPaneComponent } from '../tabs/tab-pane.component';
let TabBarItemComponent = class TabBarItemComponent extends TabPaneComponent {
    constructor() {
        super();
        this.prefixCls = 'am-tab-bar-tab';
        this.selected = false;
        this.tintColor = '#108ee9';
        this.unselectedTintColor = '#888';
        this.key = '';
        this.title = '';
        this.dot = false;
        this.badge = null;
        this.icon = null;
        this.selectedIcon = null;
    }
    isTemplateRef(value) {
        return value instanceof TemplateRef;
    }
};
__decorate([
    ViewChild('tabBarTab', { static: true }),
    __metadata("design:type", TemplateRef)
], TabBarItemComponent.prototype, "tabBarTab", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], TabBarItemComponent.prototype, "key", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], TabBarItemComponent.prototype, "title", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], TabBarItemComponent.prototype, "dot", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], TabBarItemComponent.prototype, "badge", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], TabBarItemComponent.prototype, "icon", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], TabBarItemComponent.prototype, "selectedIcon", void 0);
TabBarItemComponent = __decorate([
    Component({
        selector: 'TabBarItem, nzm-tab-bar-item',
        template: "<ng-template #content>\n  <ng-content></ng-content>\n</ng-template>\n\n<ng-template #tabBarTab>\n  <div class=\"{{ prefixCls }}-icon\" [style.color]=\"selected ? tintColor : unselectedTintColor\">\n    <Badge *ngIf=\"badge\" className=\"{{ prefixCls }}-badge tab-badge\" [text]=\"badge\">\n      <ng-container *ngIf=\"isTemplateRef(selected ? selectedIcon : icon); then domTemplate; else imgTemplate\">\n      </ng-container>\n    </Badge>\n    <Badge className=\"{{ prefixCls }}-badge tab-badge\" [dot]=\"dot\" *ngIf=\"dot\">\n      <ng-container *ngIf=\"isTemplateRef(selected ? selectedIcon : icon); then domTemplate; else imgTemplate\">\n      </ng-container>\n    </Badge>\n    <ng-container *ngIf=\"!badge && !dot\">\n      <ng-container *ngIf=\"isTemplateRef(selected ? selectedIcon : icon); then domTemplate; else imgTemplate\">\n      </ng-container>\n    </ng-container>\n  </div>\n  <p class=\"{{ prefixCls }}-title\" [style.color]=\"selected ? tintColor : unselectedTintColor\">\n    {{ title }}\n  </p>\n</ng-template>\n\n<ng-template #domTemplate>\n  <ng-template [ngTemplateOutlet]=\"selected ? selectedIcon : icon\"></ng-template>\n</ng-template>\n\n<ng-template #imgTemplate>\n  <img src=\"{{ selected ? selectedIcon : icon }}\" alt=\"{{ title }}\" />\n</ng-template>\n"
    }),
    __metadata("design:paramtypes", [])
], TabBarItemComponent);
export { TabBarItemComponent };
