// ng-zorro-antd-mobile/esm2015/input-item/custom-input/custom-input.service.js
var CustomInputService_1;
import { __decorate, __metadata } from "tslib";
import { Injectable, ComponentRef, ComponentFactory, ApplicationRef, ComponentFactoryResolver } from '@angular/core';
import { CustomKeyboardComponent } from '../custom-keyboard/custom-keyboard.component';
let CustomInputService = CustomInputService_1 = class CustomInputService {
    constructor(_appRef, _cfr) {
        this._appRef = _appRef;
        this._cfr = _cfr;
        CustomInputService_1.appRef = this._appRef;
        CustomInputService_1._inputCompFactory = this._cfr.resolveComponentFactory(CustomKeyboardComponent);
    }
    static getShowStatus() {
        return CustomInputService_1.isShow;
    }
    static showKeyboard() {
        if (!this.isShow) {
            if (this.compRef) {
                this.compRef.instance.wrapperCls = {
                    [`am-number-keyboard-wrapper`]: true
                };
            }
            else {
                let container = document.querySelector(`#${this._keyboardPrefixCls}-container`);
                if (!container) {
                    container = document.createElement('div');
                    container.setAttribute('id', `${this._keyboardPrefixCls}-container`);
                    document.body.appendChild(container);
                    container.appendChild(document.createElement(CustomInputService_1._inputCompFactory.selector));
                    this.compRef = this.appRef.bootstrap(CustomInputService_1._inputCompFactory);
                    this.compRef.instance.onClick.subscribe(e => {
                        this.clickValue = e;
                    });
                }
            }
            this.isShow = true;
        }
    }
    static hideKeyboard() {
        if (this.compRef && this.isShow) {
            this.isShow = false;
            this.compRef.instance.wrapperCls = {
                [`am-number-keyboard-wrapper`]: true,
                [`am-number-keyboard-wrapper-hide`]: true
            };
        }
    }
};
CustomInputService.compRef = null;
CustomInputService.appRef = null;
CustomInputService.isShow = false;
CustomInputService.clickValue = null;
CustomInputService._inputCompFactory = null;
CustomInputService._keyboardPrefixCls = 'am-number-keyboard';
CustomInputService.ctorParameters = () => [
    { type: ApplicationRef },
    { type: ComponentFactoryResolver }
];
CustomInputService = CustomInputService_1 = __decorate([
    Injectable(),
    __metadata("design:paramtypes", [ApplicationRef, ComponentFactoryResolver])
], CustomInputService);
export { CustomInputService };
