// ng-zorro-antd-mobile/esm2015/locale-provider/locale-provider.service.js
import { __decorate, __metadata, __param } from "tslib";
import { Inject, Injectable, Optional, SkipSelf } from '@angular/core';
import { LOCAL_PROVIDER_TOKEN } from './locale-provider.token';
import { BehaviorSubject } from 'rxjs';
import zh_CN from './locale/zh_CN';
let LocaleProviderService = class LocaleProviderService {
    constructor(locale) {
        this._change = new BehaviorSubject(this._locale);
        this.setLocale(locale || zh_CN);
    }
    get localeChange() {
        return this._change.asObservable();
    }
    getLocaleValue(keyPath) {
        const content = this._getObjectPath(this._locale, keyPath);
        if (typeof content === 'string') {
            return content;
        }
        return '';
    }
    getLocaleSubObj(keyPath) {
        const content = this._getObjectPath(this._locale, keyPath);
        if (typeof content === 'object') {
            return content;
        }
        return null;
    }
    setLocale(locale) {
        if (!locale || (this._locale && this._locale.locale === locale.locale)) {
            return;
        }
        this._locale = locale;
        this._change.next(locale);
    }
    getLocaleId() {
        return this._locale && this._locale.locale ? this._locale.locale : '';
    }
    getLocale() {
        return this._locale;
    }
    _getObjectPath(obj, path) {
        let res = obj;
        const paths = path.split('.');
        const depth = paths.length;
        let index = 0;
        while (res && index < depth) {
            res = res[paths[index++]];
        }
        return index === depth ? res : null;
    }
};
LocaleProviderService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [LOCAL_PROVIDER_TOKEN,] }] }
];
LocaleProviderService = __decorate([
    Injectable(),
    __param(0, Inject(LOCAL_PROVIDER_TOKEN)),
    __metadata("design:paramtypes", [Object])
], LocaleProviderService);
export { LocaleProviderService };
export function LOCALE_PROVIDER_SERVICE_FACTORY(exist, locale) {
    return exist || new LocaleProviderService(locale);
}
export const LOCALE_PROVIDER_SERVICE_PROVIDER = {
    provide: LocaleProviderService,
    useFactory: LOCALE_PROVIDER_SERVICE_FACTORY,
    deps: [[new Optional(), new SkipSelf(), LocaleProviderService], LOCAL_PROVIDER_TOKEN]
};
