// ng-zorro-antd-mobile/esm5/date-picker/date-picker-options.provider.js
import { __decorate } from "tslib";
import { EventEmitter, Injectable } from '@angular/core';
var DatePickerOptions = /** @class */ (function () {
    function DatePickerOptions() {
        this.mode = 'date';
        this.value = new Date();
        this.minDate = new Date(2000, 6, 1, 0, 0, 0);
        this.maxDate = new Date(2030, 1, 1, 23, 59, 59);
        this.use12Hours = false;
        this.minuteStep = 1;
        this.data = [];
        this.mask = true;
        this.title = '';
        this.okText = '确定';
        this.dismissText = '取消';
        this.disabled = false;
        this.appendToBody = false;
        this.showErrorToast = true;
        this.showErrorToastInterval = 2000;
        this.onOk = new EventEmitter();
        this.onDismiss = new EventEmitter();
        this.onValueChange = new EventEmitter();
        this.onChange = new EventEmitter();
    }
    DatePickerOptions = __decorate([
        Injectable()
    ], DatePickerOptions);
    return DatePickerOptions;
}());
export { DatePickerOptions };
