// ng-zorro-antd-mobile/esm5/carousel/dotindicator/dotindicator.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input, ViewEncapsulation, HostBinding } from '@angular/core';
var DotIndicatorComponent = /** @class */ (function () {
    function DotIndicatorComponent() {
        this.items = [];
        this._page = 0;
        this._pageCount = 0;
        this.dotStyle = {};
        this.dotActiveStyle = {};
        this.dotColor = 'white';
        this.dotIndicator = true;
    }
    Object.defineProperty(DotIndicatorComponent.prototype, "page", {
        set: function (p) {
            this._page = p;
            this.updateSelected();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DotIndicatorComponent.prototype, "pageCount", {
        set: function (p) {
            this._pageCount = p || 0;
            this.updateItems();
        },
        enumerable: true,
        configurable: true
    });
    DotIndicatorComponent.prototype.updateItems = function () {
        this.items = new Array(this._pageCount);
        for (var i = 0; i < this._pageCount; i++) {
            this.items[i] = { active: i == this._page };
        }
    };
    DotIndicatorComponent.prototype.updateSelected = function () {
        if (this.items.length != this._pageCount) {
            return this.updateItems();
        }
        if (this.items.length == 0) {
            return;
        }
        for (var i = 0; i < this._pageCount; i++) {
            this.items[i].active = false;
        }
        this.items[this._page].active = true;
    };
    __decorate([
        Input(),
        __metadata("design:type", Number),
        __metadata("design:paramtypes", [Number])
    ], DotIndicatorComponent.prototype, "page", null);
    __decorate([
        Input(),
        __metadata("design:type", Number),
        __metadata("design:paramtypes", [Number])
    ], DotIndicatorComponent.prototype, "pageCount", null);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], DotIndicatorComponent.prototype, "dotStyle", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], DotIndicatorComponent.prototype, "dotActiveStyle", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], DotIndicatorComponent.prototype, "dotColor", void 0);
    __decorate([
        HostBinding('class.dot-indicator'),
        __metadata("design:type", Boolean)
    ], DotIndicatorComponent.prototype, "dotIndicator", void 0);
    DotIndicatorComponent = __decorate([
        Component({
            selector: 'DotIndicator, nzm-dot-indicator',
            template: "<div class=\"am-carousel-wrap\">\n  <div *ngFor=\"let item of items\" class=\"am-carousel-wrap-dot\" [class.am-carousel-wrap-dot-active]=\"item.active\">\n    <span [ngStyle]=\"item.active ? dotActiveStyle : dotStyle\"></span>\n  </div>\n</div>\n",
            encapsulation: ViewEncapsulation.None
        })
    ], DotIndicatorComponent);
    return DotIndicatorComponent;
}());
export { DotIndicatorComponent };
