// ng-zorro-antd-mobile/esm2015/radio/radio-item-group.component.js
import { __decorate, __metadata } from "tslib";
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { Component, Output, AfterContentInit, OnDestroy, QueryList, forwardRef, EventEmitter, ContentChildren, ChangeDetectorRef, ChangeDetectionStrategy, ViewEncapsulation } from '@angular/core';
import { RadioItemComponent } from './radio-item.component';
import { merge, Subject } from 'rxjs';
import { startWith, takeUntil } from 'rxjs/operators';
export const RADIO_ITEM_GROUP_VALUE_ACCESSOR = {
    provide: NG_VALUE_ACCESSOR,
    useExisting: forwardRef(() => RadioItemGroupComponent),
    multi: true
};
let RadioItemGroupComponent = class RadioItemGroupComponent {
    constructor(cdr) {
        this.cdr = cdr;
        this.destroy$ = new Subject();
        this.onChange = new EventEmitter();
    }
    updateChildrenStatus() {
        if (this.radioItems && typeof this.selectedValue !== 'undefined' && null !== this.selectedValue) {
            Promise.resolve().then(() => {
                this.radioItems.forEach(radioItem => {
                    radioItem.checked = radioItem.value === this.selectedValue;
                    radioItem.markForCheck();
                });
            });
        }
    }
    ngAfterContentInit() {
        this.radioItems.changes
            .pipe(startWith(null), takeUntil(this.destroy$))
            .subscribe(() => {
            this.updateChildrenStatus();
            if (this.selectSubscription) {
                this.selectSubscription.unsubscribe();
            }
            this.selectSubscription = merge(...this.radioItems.map(radioItem => radioItem.select$))
                .pipe(takeUntil(this.destroy$))
                .subscribe(radioItem => {
                if (typeof this.selectedValue !== 'undefined' && null !== this.selectedValue) {
                    this.selectedValue = radioItem.value;
                    this._ngModelOnChange(radioItem.value);
                    this.updateChildrenStatus();
                    if (this.onChange) {
                        this.onChange.emit({ name: radioItem.name, value: radioItem.value });
                    }
                }
            });
        });
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    writeValue(value) {
        if (typeof value !== 'undefined' && null !== value) {
            this.selectedValue = value;
            this.updateChildrenStatus();
            this.cdr.markForCheck();
        }
    }
    registerOnChange(fn) {
        this._ngModelOnChange = fn;
    }
    registerOnTouched(fn) {
        this._ngModelOnTouched = fn;
    }
};
RadioItemGroupComponent.ctorParameters = () => [
    { type: ChangeDetectorRef }
];
__decorate([
    ContentChildren(forwardRef(() => RadioItemComponent)),
    __metadata("design:type", QueryList)
], RadioItemGroupComponent.prototype, "radioItems", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], RadioItemGroupComponent.prototype, "onChange", void 0);
RadioItemGroupComponent = __decorate([
    Component({
        selector: 'RadioItemGroup, nzm-radio-item-group',
        template: "<ng-content></ng-content>\n",
        encapsulation: ViewEncapsulation.None,
        changeDetection: ChangeDetectionStrategy.OnPush,
        providers: [RADIO_ITEM_GROUP_VALUE_ACCESSOR]
    }),
    __metadata("design:paramtypes", [ChangeDetectorRef])
], RadioItemGroupComponent);
export { RadioItemGroupComponent };
