// ng-zorro-antd-mobile/esm5/flex/flex.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input, ViewEncapsulation, HostBinding } from '@angular/core';
var FlexComponent = /** @class */ (function () {
    function FlexComponent() {
        this.defaultProps = {
            prefixCls: 'am-flexbox',
            align: 'center'
        };
        // _wrapCls = {};
        this._direction = '';
        this._justify = '';
        this._alignContent = '';
        this.amFlexbox = true;
    }
    Object.defineProperty(FlexComponent.prototype, "direction", {
        set: function (value) {
            this._direction = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "wrap", {
        set: function (value) {
            this._wrap = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "justify", {
        set: function (value) {
            this._justify = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "align", {
        set: function (value) {
            this.defaultProps.align = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "alignContent", {
        set: function (value) {
            this._alignContent = value;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxDirRow", {
        get: function () {
            return this._direction === 'row';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxDirRowReverse", {
        get: function () {
            return this._direction === 'row-reverse';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxDirColumn", {
        get: function () {
            return this._direction === 'column';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxDirColumnReverse", {
        get: function () {
            return this._direction === 'column-reverse';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxNowrap", {
        get: function () {
            return this._wrap === 'nowrap';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxWrap", {
        get: function () {
            return this._wrap === 'wrap';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxWrapReverse", {
        get: function () {
            return this._wrap === 'wrap-reverse';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxJustifyStart", {
        get: function () {
            return this._justify === 'start';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxJustifyCenter", {
        get: function () {
            return this._justify === 'center';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxJustifyEnd", {
        get: function () {
            return this._justify === 'end';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxJustifyBetween", {
        get: function () {
            return this._justify === 'between';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignAround", {
        get: function () {
            return this._justify === 'around';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignStart", {
        get: function () {
            return this.defaultProps.align === 'start';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignCenter", {
        get: function () {
            return this.defaultProps.align === 'center';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignEnd", {
        get: function () {
            return this.defaultProps.align === 'end';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignBaseline", {
        get: function () {
            return this.defaultProps.align === 'baseline';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignStretch", {
        get: function () {
            return this.defaultProps.align === 'stretch';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignContentStart", {
        get: function () {
            return this._alignContent === 'start';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignCotentCenter", {
        get: function () {
            return this._alignContent === 'center';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignContentEnd", {
        get: function () {
            return this._alignContent === 'end';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignContentBetween", {
        get: function () {
            return this._alignContent === 'between';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignContentAround", {
        get: function () {
            return this._alignContent === 'around';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(FlexComponent.prototype, "amFlexboxAlignContentStretch", {
        get: function () {
            return this._alignContent === 'stretch';
        },
        enumerable: true,
        configurable: true
    });
    __decorate([
        Input(),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [Object])
    ], FlexComponent.prototype, "direction", null);
    __decorate([
        Input(),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [Object])
    ], FlexComponent.prototype, "wrap", null);
    __decorate([
        Input(),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [Object])
    ], FlexComponent.prototype, "justify", null);
    __decorate([
        Input(),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [Object])
    ], FlexComponent.prototype, "align", null);
    __decorate([
        Input(),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [Object])
    ], FlexComponent.prototype, "alignContent", null);
    __decorate([
        HostBinding('class.am-flexbox'),
        __metadata("design:type", Boolean)
    ], FlexComponent.prototype, "amFlexbox", void 0);
    __decorate([
        HostBinding('class.am-flexbox-dir-row'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxDirRow", null);
    __decorate([
        HostBinding('class.am-flexbox-dir-row-reverse'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxDirRowReverse", null);
    __decorate([
        HostBinding('class.am-flexbox-dir-column'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxDirColumn", null);
    __decorate([
        HostBinding('class.am-flexbox-dir-column-reverse'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxDirColumnReverse", null);
    __decorate([
        HostBinding('class.am-flexbox-nowrap'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxNowrap", null);
    __decorate([
        HostBinding('class.am-flexbox-wrap'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxWrap", null);
    __decorate([
        HostBinding('class.am-flexbox-wrap-reverse'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxWrapReverse", null);
    __decorate([
        HostBinding('class.am-flexbox-justify-start'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxJustifyStart", null);
    __decorate([
        HostBinding('class.am-flexbox-justify-center'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxJustifyCenter", null);
    __decorate([
        HostBinding('class.am-flexbox-justify-end'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxJustifyEnd", null);
    __decorate([
        HostBinding('class.am-flexbox-justify-between'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxJustifyBetween", null);
    __decorate([
        HostBinding('class.am-flexbox-justify-around'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignAround", null);
    __decorate([
        HostBinding('class.am-flexbox-align-start'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignStart", null);
    __decorate([
        HostBinding('class.am-flexbox-align-center'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignCenter", null);
    __decorate([
        HostBinding('class.am-flexbox-align-end'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignEnd", null);
    __decorate([
        HostBinding('class.am-flexbox-align-baseline'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignBaseline", null);
    __decorate([
        HostBinding('class.am-flexbox-align-stretch'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignStretch", null);
    __decorate([
        HostBinding('class.am-flexbox-align-content-start'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignContentStart", null);
    __decorate([
        HostBinding('class.am-flexbox-align-content-center'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignCotentCenter", null);
    __decorate([
        HostBinding('class.am-flexbox-align-content-end'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignContentEnd", null);
    __decorate([
        HostBinding('class.am-flexbox-align-content-between'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignContentBetween", null);
    __decorate([
        HostBinding('class.am-flexbox-align-content-around'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignContentAround", null);
    __decorate([
        HostBinding('class.am-flexbox-align-content-stretch'),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [])
    ], FlexComponent.prototype, "amFlexboxAlignContentStretch", null);
    FlexComponent = __decorate([
        Component({
            selector: 'Flex, nzm-flex',
            template: "<ng-content></ng-content>\n",
            encapsulation: ViewEncapsulation.None
        }),
        __metadata("design:paramtypes", [])
    ], FlexComponent);
    return FlexComponent;
}());
export { FlexComponent };
var FlexItemComponent = /** @class */ (function () {
    function FlexItemComponent() {
        this.defaultProps = {
            prefixCls: 'am-flexbox',
            align: 'center'
        };
        this.flexboxItem = true;
    }
    __decorate([
        HostBinding('class.am-flexbox-item'),
        __metadata("design:type", Boolean)
    ], FlexItemComponent.prototype, "flexboxItem", void 0);
    FlexItemComponent = __decorate([
        Component({
            selector: 'FlexItem, nzm-flex-item',
            template: "\n    <ng-content></ng-content>\n  ",
            encapsulation: ViewEncapsulation.None
        })
    ], FlexItemComponent);
    return FlexItemComponent;
}());
export { FlexItemComponent };
