// ng-zorro-antd-mobile/esm5/core/services/popup.service.js
import { __decorate, __metadata } from "tslib";
import { Injectable } from '@angular/core';
import { Overlay, OverlayConfig, OverlayRef, GlobalPositionStrategy } from '@angular/cdk/overlay';
import { ComponentPortal } from '@angular/cdk/portal';
var PopupService = /** @class */ (function () {
    function PopupService(_overlay) {
        this._overlay = _overlay;
        this.overlay = null;
        this.overlayRef = null;
        this.currentServiceName = null;
        this.serviceArray = [];
        this.overlay = this._overlay;
    }
    PopupService.prototype.showPopup = function (component, childInjector, hasBackdrop, positionStrategy) {
        var _this = this;
        if (positionStrategy === void 0) { positionStrategy = this.overlay
            .position()
            .global()
            .centerVertically()
            .centerHorizontally(); }
        var overlayConfig = new OverlayConfig();
        overlayConfig.hasBackdrop = hasBackdrop;
        overlayConfig.positionStrategy = positionStrategy;
        this.overlayRef = this.overlay.create(overlayConfig);
        this.overlayRef.backdropClick().subscribe(function () {
            _this.hidePopup();
        });
        return this.overlayRef.attach(new ComponentPortal(component, undefined, childInjector));
    };
    PopupService.prototype.hidePopup = function () {
        if (this.overlayRef) {
            this.overlayRef.dispose();
        }
    };
    PopupService.ctorParameters = function () { return [
        { type: Overlay }
    ]; };
    PopupService = __decorate([
        Injectable({
            providedIn: 'root'
        }),
        Injectable(),
        __metadata("design:paramtypes", [Overlay])
    ], PopupService);
    return PopupService;
}());
export { PopupService };
