// ng-zorro-antd-mobile/esm5/tabs/tab-pane-body.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input, HostBinding, TemplateRef, ViewEncapsulation } from '@angular/core';
var TabPaneBodyComponent = /** @class */ (function () {
    function TabPaneBodyComponent() {
        this._prerender = false;
        this.active = false;
        this.loaded = false;
        this.paneWrap = true;
    }
    Object.defineProperty(TabPaneBodyComponent.prototype, "prerender", {
        get: function () {
            return this._prerender;
        },
        set: function (value) {
            this._prerender = value;
            if (value) {
                this.loaded = true;
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TabPaneBodyComponent.prototype, "wrapActive", {
        get: function () {
            return this.active;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(TabPaneBodyComponent.prototype, "wrapInactive", {
        get: function () {
            return !this.active;
        },
        enumerable: true,
        configurable: true
    });
    TabPaneBodyComponent.prototype.ngOnInit = function () { };
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], TabPaneBodyComponent.prototype, "active", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], TabPaneBodyComponent.prototype, "loaded", void 0);
    __decorate([
        Input(),
        __metadata("design:type", TemplateRef)
    ], TabPaneBodyComponent.prototype, "content", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean),
        __metadata("design:paramtypes", [Boolean])
    ], TabPaneBodyComponent.prototype, "prerender", null);
    __decorate([
        HostBinding('class.am-tabs-pane-wrap'),
        __metadata("design:type", Boolean)
    ], TabPaneBodyComponent.prototype, "paneWrap", void 0);
    __decorate([
        HostBinding('class.am-tabs-pane-wrap-active'),
        __metadata("design:type", Boolean),
        __metadata("design:paramtypes", [])
    ], TabPaneBodyComponent.prototype, "wrapActive", null);
    __decorate([
        HostBinding('class.am-tabs-pane-wrap-inactive'),
        __metadata("design:type", Boolean),
        __metadata("design:paramtypes", [])
    ], TabPaneBodyComponent.prototype, "wrapInactive", null);
    TabPaneBodyComponent = __decorate([
        Component({
            selector: '[tab-pane-body]',
            template: "<ng-container *ngIf=\"loaded || prerender\">\n  <ng-template [ngTemplateOutlet]=\"content\"></ng-template>\n</ng-container>\n",
            encapsulation: ViewEncapsulation.None
        }),
        __metadata("design:paramtypes", [])
    ], TabPaneBodyComponent);
    return TabPaneBodyComponent;
}());
export { TabPaneBodyComponent };
