// ng-zorro-antd-mobile/esm5/carousel/carousel-slide/carousel-slide.component.js
import { __decorate, __metadata } from "tslib";
import { Component, HostBinding, ViewEncapsulation } from '@angular/core';
var CarouselSlideComponent = /** @class */ (function () {
    function CarouselSlideComponent() {
        this.container = true;
        this.height = 'auto';
        this.overflow = 'hidden';
    }
    __decorate([
        HostBinding('class.am-carousel-container'),
        __metadata("design:type", Object)
    ], CarouselSlideComponent.prototype, "container", void 0);
    __decorate([
        HostBinding('style.width.px'),
        __metadata("design:type", Object)
    ], CarouselSlideComponent.prototype, "width", void 0);
    __decorate([
        HostBinding('style.height'),
        __metadata("design:type", Object)
    ], CarouselSlideComponent.prototype, "height", void 0);
    __decorate([
        HostBinding('style.left.px'),
        __metadata("design:type", Object)
    ], CarouselSlideComponent.prototype, "left", void 0);
    __decorate([
        HostBinding('style.top.px'),
        __metadata("design:type", Object)
    ], CarouselSlideComponent.prototype, "top", void 0);
    __decorate([
        HostBinding('style.margin'),
        __metadata("design:type", Object)
    ], CarouselSlideComponent.prototype, "margin", void 0);
    __decorate([
        HostBinding('style.overflow'),
        __metadata("design:type", Object)
    ], CarouselSlideComponent.prototype, "overflow", void 0);
    CarouselSlideComponent = __decorate([
        Component({
            selector: 'CarouselSlide, nzm-carousel-slide',
            template: "<ng-content></ng-content>\n",
            encapsulation: ViewEncapsulation.None
        })
    ], CarouselSlideComponent);
    return CarouselSlideComponent;
}());
export { CarouselSlideComponent };
