// ng-zorro-antd-mobile/esm5/menu/menu.component.js
import { __decorate, __metadata, __read, __spread } from "tslib";
import { Component, ViewEncapsulation, Input, Output, EventEmitter } from '@angular/core';
import { Subject } from 'rxjs';
import { LocaleProviderService } from '../locale-provider/locale-provider.service';
import { takeUntil } from 'rxjs/operators';
var MenuComponent = /** @class */ (function () {
    function MenuComponent(_localeProviderService) {
        this._localeProviderService = _localeProviderService;
        this.prefixCls = 'am-menu';
        this.subMenuPrefixCls = 'am-sub-menu';
        this.radioPrefixCls = 'am-radio';
        this.multiSelectMenuBtnsCls = 'am-multi-select-btns';
        this.menuSelectContanerPrefixCls = 'am-menu-select-container';
        this.locale = {
            okText: '',
            cancelText: ''
        };
        this._data = [];
        this._unsubscribe$ = new Subject();
        this.level = 2;
        this.value = [];
        this.height = document.documentElement.clientHeight / 2;
        this.multiSelect = false;
        this.onChange = new EventEmitter();
        this.onOk = new EventEmitter();
        this.onCancel = new EventEmitter();
    }
    Object.defineProperty(MenuComponent.prototype, "data", {
        get: function () {
            return this._data;
        },
        set: function (v) {
            this._data = v;
            this.initData();
        },
        enumerable: true,
        configurable: true
    });
    MenuComponent.prototype.onMenuOk = function () {
        this.onOk.emit(this.value);
    };
    MenuComponent.prototype.onMenuCancel = function () {
        this.onCancel.emit();
    };
    MenuComponent.prototype.getNewFsv = function () {
        var firstValue = '';
        if (this.value && this.value.length) {
            firstValue = this.value[0];
        }
        else if (this._data && this._data.length && !this._data[0].isLeaf) {
            firstValue = this._data[0].value;
        }
        return firstValue;
    };
    MenuComponent.prototype.onClickFirstLevelItem = function (dataItem) {
        this.firstLevelSelectValue = dataItem.value;
        if (dataItem.isLeaf && this.onChange) {
            this.onChange.emit([dataItem.value]);
        }
        this.initData();
    };
    MenuComponent.prototype.onClickSubMenuItem = function (dataItem) {
        var _this = this;
        this.value = this.getSelectValue(dataItem);
        this.initData();
        setTimeout(function () {
            _this.onChange.emit(_this.value);
        }, 300);
    };
    MenuComponent.prototype.getSelectValue = function (dataItem) {
        if (this.multiSelect) {
            if (this.value && this.value.length > 0) {
                if (this.level === 2 && this.value[0] !== this.firstLevelSelectValue) {
                    return [this.firstLevelSelectValue, [dataItem.value]];
                }
                else {
                    if (this.level == 1) {
                        var chosenValues = Array.from(this.value);
                        var existIndex = chosenValues.indexOf(dataItem.value);
                        if (existIndex === -1) {
                            chosenValues.push(dataItem.value);
                        }
                        else {
                            chosenValues.splice(existIndex, 1);
                        }
                        return chosenValues;
                    }
                    else {
                        var chosenValues = Array.from(this.value[1]);
                        var existIndex = chosenValues.indexOf(dataItem.value);
                        if (existIndex === -1) {
                            chosenValues.push(dataItem.value);
                        }
                        else {
                            chosenValues.splice(existIndex, 1);
                        }
                        return [this.firstLevelSelectValue, chosenValues];
                    }
                }
            }
            else {
                return this.level === 2 ? [this.firstLevelSelectValue, [dataItem.value]] : [dataItem.value];
            }
        }
        return this.level === 2 ? [this.firstLevelSelectValue, dataItem.value] : [dataItem.value];
    };
    MenuComponent.prototype.initData = function () {
        var _this = this;
        this.subMenuData = this._data;
        if (this.level === 2) {
            var parent_1 = this._data;
            if (this.firstLevelSelectValue && this.firstLevelSelectValue !== '') {
                parent_1 = this._data.filter(function (dataItem) { return dataItem.value === _this.firstLevelSelectValue; });
            }
            if (parent_1[0] && parent_1[0].children && parent_1[0].isLeaf !== true) {
                this.subMenuData = parent_1[0].children;
            }
            else {
                this.subMenuData = [];
            }
        }
        var subValue = (this.value && this.value.length > 0 && __spread(this.value)) || [];
        if (this.level === 2 && subValue.length > 1) {
            subValue.shift();
            if (this.multiSelect) {
                subValue = subValue[0];
            }
        }
        this.subSelInitItem = this.subMenuData
            .filter(function (dataItem) { return subValue.indexOf(dataItem.value) !== -1; })
            .map(function (item) {
            return item.value;
        });
        var parentValue = this.value && this.value.length > 1 && this.level === 2 ? this.value[0] : null;
        this.showSelect = true;
        if (this.level === 2 && parentValue !== this.firstLevelSelectValue) {
            this.showSelect = false;
        }
    };
    MenuComponent.prototype.getClass = function (dataItem) {
        return this.dataItemSelected(dataItem) ? this.prefixCls + '-selected' : '';
    };
    MenuComponent.prototype.dataItemSelected = function (dataItem) {
        return dataItem.value === this.firstLevelSelectValue;
    };
    MenuComponent.prototype.ngOnInit = function () {
        var _this = this;
        this._localeProviderService.localeChange.pipe(takeUntil(this._unsubscribe$)).subscribe(function (_) {
            _this.locale = _this._localeProviderService.getLocaleSubObj('Menu');
        });
        this.firstLevelSelectValue = this.getNewFsv();
        this.heightStyle = {
            height: this.height + 'px'
        };
        this.initData();
    };
    MenuComponent.prototype.ngOnDestroy = function () {
        this._unsubscribe$.next();
        this._unsubscribe$.complete();
    };
    MenuComponent.ctorParameters = function () { return [
        { type: LocaleProviderService }
    ]; };
    __decorate([
        Input(),
        __metadata("design:type", Object),
        __metadata("design:paramtypes", [Object])
    ], MenuComponent.prototype, "data", null);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], MenuComponent.prototype, "level", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Array)
    ], MenuComponent.prototype, "value", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Number)
    ], MenuComponent.prototype, "height", void 0);
    __decorate([
        Input(),
        __metadata("design:type", Boolean)
    ], MenuComponent.prototype, "multiSelect", void 0);
    __decorate([
        Output(),
        __metadata("design:type", EventEmitter)
    ], MenuComponent.prototype, "onChange", void 0);
    __decorate([
        Output(),
        __metadata("design:type", EventEmitter)
    ], MenuComponent.prototype, "onOk", void 0);
    __decorate([
        Output(),
        __metadata("design:type", EventEmitter)
    ], MenuComponent.prototype, "onCancel", void 0);
    MenuComponent = __decorate([
        Component({
            selector: 'Menu, nzm-menu',
            template: "<Flex class=\"{{ prefixCls }}\" [ngStyle]=\"heightStyle\" [direction]=\"'column'\" [align]=\"'stretch'\">\n  <Flex class=\"{{ menuSelectContanerPrefixCls }}\" [align]=\"'start'\">\n    <FlexItem *ngIf=\"level == 2\">\n      <List role=\"tablist\">\n        <ListItem\n          role=\"tab\"\n          *ngFor=\"let dataItem of data; let i = index\"\n          [className]=\"getClass(dataItem)\"\n          (click)=\"onClickFirstLevelItem(dataItem)\"\n        >\n          {{ dataItem.label }}\n        </ListItem>\n      </List>\n    </FlexItem>\n\n    <FlexItem role=\"tabpanel\" aria-hidden=\"false\" class=\"{{ menuSelectContanerPrefixCls }}-submenu\">\n      <SubMenu\n        [subMenuPrefixCls]=\"subMenuPrefixCls\"\n        [radioPrefixCls]=\"radioPrefixCls\"\n        [subMenuData]=\"subMenuData\"\n        [selItem]=\"subSelInitItem\"\n        [showSelect]=\"showSelect\"\n        [multiSelect]=\"multiSelect\"\n        (onSel)=\"onClickSubMenuItem($event)\"\n      >\n      </SubMenu>\n    </FlexItem>\n  </Flex>\n\n  <div *ngIf=\"multiSelect\" class=\"{{ multiSelectMenuBtnsCls }}\">\n    <a Button [className]=\"'am-multi-select-btns-btn'\" [inline]=\"true\" (onClick)=\"onMenuCancel()\">\n      {{ locale.cancelText }}\n    </a>\n    <a Button [className]=\"'am-multi-select-btns-btn'\" [inline]=\"true\" [type]=\"'primary'\" (onClick)=\"onMenuOk()\">\n      {{ locale.okText }}\n    </a>\n  </div>\n</Flex>\n",
            encapsulation: ViewEncapsulation.None
        }),
        __metadata("design:paramtypes", [LocaleProviderService])
    ], MenuComponent);
    return MenuComponent;
}());
export { MenuComponent };
