// ng-zorro-antd-mobile/esm2015/popover/popover-item/popover-item.component.js
import { __decorate, __metadata } from "tslib";
import { Component, AfterContentInit, Input, Output, ElementRef, TemplateRef, HostBinding, HostListener, EventEmitter, ViewEncapsulation } from '@angular/core';
let PopoverItemComponent = class PopoverItemComponent {
    constructor(_elementRef) {
        this._elementRef = _elementRef;
        this.defaultProps = {
            prefixCls: 'am-popover',
            disabled: false
        };
        this.isActive = false;
        this.select = new EventEmitter();
        this.amPopoverItem = true;
    }
    get icon() {
        return this._icon;
    }
    set icon(value) {
        this._icon = value;
    }
    get style() {
        return this._style;
    }
    set style(value) {
        this._style = value;
    }
    set disabled(value) {
        this.defaultProps.disabled = value;
    }
    get amPopoverItemActive() {
        return this.isActive;
    }
    get amPopoverItemDisabled() {
        return this.defaultProps.disabled;
    }
    touchStart(e) {
        this.select.emit();
        this.isActive = true;
    }
    ngAfterContentInit() { }
};
PopoverItemComponent.ctorParameters = () => [
    { type: ElementRef }
];
__decorate([
    Input(),
    __metadata("design:type", TemplateRef),
    __metadata("design:paramtypes", [TemplateRef])
], PopoverItemComponent.prototype, "icon", null);
__decorate([
    Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], PopoverItemComponent.prototype, "style", null);
__decorate([
    Input(),
    __metadata("design:type", Object),
    __metadata("design:paramtypes", [Object])
], PopoverItemComponent.prototype, "disabled", null);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], PopoverItemComponent.prototype, "select", void 0);
__decorate([
    HostBinding('class.am-popover-item'),
    __metadata("design:type", Boolean)
], PopoverItemComponent.prototype, "amPopoverItem", void 0);
__decorate([
    HostBinding('class.am-popover-item-active'),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [])
], PopoverItemComponent.prototype, "amPopoverItemActive", null);
__decorate([
    HostBinding('class.am-popover-item-disabled'),
    __metadata("design:type", Boolean),
    __metadata("design:paramtypes", [])
], PopoverItemComponent.prototype, "amPopoverItemDisabled", null);
__decorate([
    HostListener('touchstart', ['$event']),
    HostListener('mousedown', ['$event']),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", [Object]),
    __metadata("design:returntype", void 0)
], PopoverItemComponent.prototype, "touchStart", null);
PopoverItemComponent = __decorate([
    Component({
        selector: 'PopoverItem, nzm-popover-item',
        template: "<div class=\"{{ defaultProps.prefixCls }}-item-container\">\n  <span class=\"{{ defaultProps.prefixCls }}-item-icon\" aria-hidden=\"true\">\n    <ng-template [ngTemplateOutlet]=\"icon\"></ng-template>\n  </span>\n  <span class=\"{{ defaultProps.prefixCls }}-item-content\">\n    <ng-content></ng-content>\n  </span>\n</div>\n",
        encapsulation: ViewEncapsulation.None
    }),
    __metadata("design:paramtypes", [ElementRef])
], PopoverItemComponent);
export { PopoverItemComponent };
